scripts/irods/test/test_itouch.py

```diff
@@ -0,0 +1,216 @@
+
+            # Make the first data object's mtime match the second data object's mtime.
+            # This also demonstrates that the reference path is relative to the current
+            # working collection.
+            self.admin.assert_icommand(['itouch', '-r', data_object_2, data_object_1])
+            self.assertEqual(self.get_replica_mtime(data_object_1, 0), mtime)
+        except Exception as e:
+            print(e)
+
+    @unittest.skipIf(test.settings.RUN_IN_TOPOLOGY, "Skip for Topology Testing")
+    def test_seconds_since_epoch_is_honored(self):
+        data_object = 'foo'
+
+        # 584582400 seconds since epoch => 1988-07-11 00:00:00.0 UTC
+        #                               => 1988-07-10 20:00:00.0 Local Time
+        self.admin.assert_icommand(['itouch', '-s', '584582400', data_object])
+        self.admin.assert_icommand(['ils', '-l', data_object], 'STDOUT', ['0 1988-07-10.20:00 &'])
+
+    @unittest.skipIf(test.settings.RUN_IN_TOPOLOGY, "Skip for Topology Testing")
+    def test_collections_are_supported(self):
+        try:
+            old_mtime = self.get_collection_mtime(self.admin.session_collection)
+            sleep(2)
+            self.admin.assert_icommand(['itouch', self.admin.session_collection])
+            self.assertNotEqual(old_mtime, self.get_collection_mtime(self.admin.session_collection))
+        except Exception as e:
+            print(e)
+
+    @unittest.skipIf(test.settings.RUN_IN_TOPOLOGY, "Skip for Topology Testing")
+    def test_mutually_exclusive_options_are_handled(self):
+        error_msg = 'Error: -n and -R cannot be used together.'
+        self.admin.assert_icommand(['itouch', '-n0', '-R', 'demoResc', 'foo'], 'STDERR', [error_msg])
+
+        error_msg = 'Error: -r and -s cannot be used together.'
+        self.admin.assert_icommand(['itouch', '-r', 'source', '-s', '1', 'foo'], 'STDERR', [error_msg])
+
+    @unittest.skipIf(test.settings.RUN_IN_TOPOLOGY, "Skip for Topology Testing")
+    def test_replica_number_cannot_be_used_to_create_new_replicas(self):
+        error_msg = 'Error: Replica numbers cannot be used when creating new data objects.'
+        self.admin.assert_icommand(['itouch', '-n3', 'foo'], 'STDERR', [error_msg])
+
+    def get_replica_mtime(self, path, replica_number):
+        collection = os.path.dirname(path)
+        if len(collection) == 0:
+            collection = self.admin.session_collection
+        data_object = os.path.basename(path)
+        mtime, ec, rc = self.admin.run_icommand('iquest %s "select DATA_MODIFY_TIME where COLL_NAME = \'{0}\' and DATA_NAME = \'{1}\' and DATA_REPL_NUM = \'{2}\'"'
+            .format(collection, data_object, replica_number))
+        return int(mtime)
+
+    def get_collection_mtime(self, path):
+        mtime, ec, rc = self.admin.run_icommand('iquest %s "select COLL_MODIFY_TIME where COLL_NAME = \'{0}\'"'.format(path))
+        return int(mtime)
+
+    def create_resource(self, resource_name):
```

Comment:
this is fine... but we may already have a helper for this - not sure...

Reply:
i'm going to leave it for now.  the tests are passing and i don't want to hold this up at this time.